### ISSUE TITLE
dnf: RFE: Implement dependency querying (pk_backend_depends_on, pk_backend_required_by)

### ISSUE THREAD

**Issue by kkofler** [PackageKit/PackageKit#106]
Apper can show this information if the backend provides it. (E.g., the old yum backend provided it.)

The recursive case is fairly complicated (does anybody actually use that?), but the non-recursive versions should be enough for Apper. There, I would just expect a full unfiltered list of what amounts to `repoquery --requires foo` resp. `repoquery --whatrequires --alldeps foo`. The first is just a `hy_package_get_requires`. For the second, we need a `hy_query_filter_reldep_in(…, HY_PKG_REQUIRES, hy_package_get_provides(…))`.

**Comment by Conan-Kudo** [comment]
Note that Mageia's urpmi backend also supports this feature, and it would be great for it to be properly supported for a migration from the urpmi backend to the hif backend as the default for PackageKit in Mageia 6.

**Comment by hughsie** [comment]
Right, I'm not against providing the functionality but I really don't really have time to write the code. If somebody does a PR I'm happy to review and commit it.

**Comment by Conan-Kudo** [comment]
I started trying to implement this a while ago in #495, I'm hoping I can get this implemented soon...

**Comment by sidt4** [comment]
`hif` backend is superseded by the `dnf`.

If this is still relevant to `dnf` backend, we can update the issue title. Else we can close this.
